src/Bicep.Core/SourceCode/SourceArchive.cs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using Bicep.Core.Semantics;
+using Bicep.Core.Workspaces;
+using System.Text.Json;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization.Metadata;
+using System.Text;
+using Bicep.Core.Navigation;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using System.Collections.Immutable;
+using static Bicep.Core.SourceCode.SourceArchive;
+using System.Text.Json.Serialization;
+using Bicep.Core.Registry.Oci;
+using Bicep.Core.Exceptions;
+
+namespace Bicep.Core.SourceCode
+{
+    // Contains the individual source code files for a Bicep file and all of its dependencies.
+    public partial class SourceArchive
```

Comment:
>partial [](http://example.com/codeflow?start=11&length=7)

No issue, but I'm curious why this is a partial class.

Reply:
// Contains the individual source code files for a Bicep file and all of its dependencies.
    // (has to be partial for dotnet to generate code for JsonSerializable nested classes
    public partial class SourceArchive